sale_order_wh_from_delivery_carrier/models/sale_order.py

```diff
@@ -0,0 +1,21 @@
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends(
+        "user_id", "company_id", "state", "carrier_id", "carrier_id.so_warehouse_id"
```

Comment:
You don't need to trigger recomputation on state change or if you reconfigure the carrier
```suggestion
        "user_id", "company_id", "carrier_id"
```